src/internal/tabbable.test.ts

```diff
@@ -145,3 +141,36 @@ it('Should allow tabbing to slotted elements', async () => {
   await holdShiftKey(async () => await sendKeys({ press: tabKey }));
   expect(activeElementsArray()).to.include(focusSix);
 });
+
+it('Should account for when focus is changed from outside sources (like clicking)', async () => {
+  const dialog = await fixture(html`
+    <sl-dialog open="" label="Dialog" class="dialog-overview">
+      Lorem ipsum dolor sit amet, consectetur adipiscing elit.
+      <sl-input placeholder="tab to me"></sl-input>
+      <sl-button slot="footer" variant="primary">Close</sl-button>
+    </sl-dialog>
+  `);
+
+  const inputEl = dialog.querySelector('sl-input')!;
+  const closeButton = dialog.shadowRoot!.querySelector('sl-icon-button')!;
+  const footerButton = dialog.querySelector('sl-button')!;
+
+  expect(activeElementsArray()).to.not.include(inputEl);
+
+  // Sets focus to the input element
+  inputEl.focus();
+
+  expect(activeElementsArray()).to.include(inputEl);
+
+  await sendKeys({ press: tabKey });
+
+  expect(activeElementsArray()).not.to.include(inputEl);
+  expect(activeElementsArray()).to.include(footerButton);
```

Comment:
Perfect :+1:

(before this change, hitting tab would move focus "backwards" to the close button)